File: 프로그래머스/2/154539. 뒤에 있는 큰 수 찾기/뒤에 있는 큰 수 찾기.cpp
#include <string>
#include <vector>
#include <stack>

using namespace std;

vector<int> solution(vector<int> numbers) {
    int len = numbers.size();
    vector<int> answer(len, -1);
    stack<int> idxs;
    
    for(int i =0; i<len ;i++)
    {
        while(!idxs.empty())
        {
            if(numbers[idxs.top()] >= numbers[i]) break;
            else
            {
                answer[idxs.top()] = numbers[i];
                idxs.pop();
            }
        }
        idxs.push(i);
    }
    
    
    return answer;
}

File: 프로그래머스/2/154539. 뒤에 있는 큰 수 찾기/뒤에 있는 큰 수 찾기.js
function solution(numbers) {
    const len = numbers.length;
    var answer = Array.from({length :len}, () => -1);
    
    let arr = [];
    
    for(let i=0; i<len; i++) {
        
        while(arr.length !== 0) {
            if(numbers[arr[arr.length-1]] >= numbers[i]) break;
            else {
                answer[arr[arr.length-1]] = numbers[i];
                arr.pop();
            }
        }
        
        arr.push(i);
    }
    
    return answer;
}